llvm/include/llvm/Object/ELFObjectFile.h

```diff
@@ -1453,6 +1525,15 @@ template <class ELFT> bool ELFObjectFile<ELFT>::isRelocatableObject() const {
   return EF.getHeader().e_type == ELF::ET_REL;
 }
 
+template <class ELFT>
+StringRef ELFObjectFile<ELFT>::getCrelError(DataRefImpl Sec) const {
+  uintptr_t SHT = reinterpret_cast<uintptr_t>(cantFail(EF.sections()).begin());
+  auto I = (Sec.p - SHT) / EF.getHeader().e_shentsize;
+  if (I < CrelErrs.size())
+    return CrelErrs[I];
+  return "";
```

Comment:
Wouldn't this be an error (or perhaps internal error) for an out of bounds access?

Reply:
I have changed `mutable SmallVector<std::string, 0> CrelErrs;` so that it is only non-empty when there is a decoder problem. This is now needed.